components/reference-data/src/main/java/org/datacleaner/beans/transform/RemoveDictionaryMatchesTransformer.java

```diff
@@ -175,9 +194,26 @@ public void close() {
                     }
 
                     value = value.substring(0, start) + value.substring(end);
-                    removedParts.add(entry);
+                    removedParts.add(entry.getKey());
+                }
+            }
+            
+            // do word-by-word dictionary lookups
```

Comment:
Ahhh! Misunderstood in first overview. This doesn't change the entire working method when a multiword entry is found, but is only different for those few words.

That makes it much better. My worry was that the working method of everything would change as soon as a multi word was inserted into the dictionary.

In reality, if there is any difference (besides the wanted difference), it is limited to those entries, which is much, much less critical. :+1: